js/graph_legend.js

```diff
@@ -128,24 +137,44 @@ function GraphLegend(targetElem, config, onVisibilityChange, onNewSelectionChang
         if(!config.selectedFieldName) $('.hide-analyser-window').hide();
     }
 
-    this.updateValues = function(flightLog, frame) {
-      try {  
-          // New function to show values on legend.
-          var currentFlightMode = frame[flightLog.getMainFieldIndexByName("flightModeFlags")];
-          
-              $(".graph-legend-field").each(function(index, value) {
-                  var value = frame[flightLog.getMainFieldIndexByName($(this).attr('name'))]; // get the raw value from log
-                  if(userSettings.legendUnits) { // if we want the legend to show engineering units
-                      value = FlightLogFieldPresenter.decodeFieldToFriendly(flightLog, $(this).attr('name'), value, currentFlightMode);
-                  } else { // raw value
-                    if(value%1!=0) { value = value.toFixed(2); }
-                  }
-                  $(this).text(FlightLogFieldPresenter.fieldNameToFriendly($(this).attr('name'), flightLog.getSysConfig().debug_mode) + ((value!=null)?' (' + value + ')':' ') );
-                  $(this).append('<span class="glyphicon glyphicon-equalizer no-wheel"></span>');
-              });
-          } catch(e) {
-              console.log('Cannot update legend with values');
-          }
+    this.updateValues = function (flightLog, frame) {
+        try {
+            // New function to show values on legend.
+            var currentFlightMode = frame[ flightLog.getMainFieldIndexByName("flightModeFlags") ];
+            var
+                graphs = config.getGraphs(),
+                i, j;
+
+            $(".graph-legend-field-value").each(function (index, value) {
+                var fieldName = $(this).attr('name');
+                var value = frame[ flightLog.getMainFieldIndexByName(fieldName) ]; // get the raw value from log
+                if (userSettings.legendUnits) { // if we want the legend to show engineering units
+                    value = FlightLogFieldPresenter.decodeFieldToFriendly(flightLog, fieldName, value, currentFlightMode);
+                } else { // raw value
+                    if (value % 1 != 0) { value = value.toFixed(2); }
```

Comment:
Not in the same line.

Reply:
Oh, I more or less just changed the indentation on that section. But I'll fix it! :)